presto-main/src/main/java/com/facebook/presto/execution/buffer/SerializedPage.java

```diff
@@ -16,31 +16,32 @@
 import io.airlift.slice.Slice;
 import org.openjdk.jol.info.ClassLayout;
 
-import static com.facebook.presto.execution.buffer.PageCompression.COMPRESSED;
-import static com.facebook.presto.execution.buffer.PageCompression.UNCOMPRESSED;
 import static com.google.common.base.MoreObjects.toStringHelper;
 import static com.google.common.base.Preconditions.checkArgument;
 import static java.util.Objects.requireNonNull;
 
 public class SerializedPage
 {
     private static final int INSTANCE_SIZE = ClassLayout.parseClass(SerializedPage.class).instanceSize();
-    private static final int PAGE_COMPRESSION_SIZE = ClassLayout.parseClass(PageCompression.class).instanceSize();
 
     private final Slice slice;
-    private final PageCompression compression;
     private final int positionCount;
     private final int uncompressedSizeInBytes;
+    private final byte pageCodecMarkings;
 
-    public SerializedPage(Slice slice, PageCompression compression, int positionCount, int uncompressedSizeInBytes)
+    public SerializedPage(Slice slice, byte pageCodecMarkings, int positionCount, int uncompressedSizeInBytes)
```

Comment:
I'm not sure, since we aren't really identifying a single "Codec" but instead just boolean properties of the way the page was encoded / should be decoded. Would `PageMarker` be better? I agree that `PageCodecMarker` is a little too verbose, but I failed to come up with something better myself.